spec/unit/actions/synchronize_users_spec.rb

```diff
@@ -130,5 +130,52 @@
         let(:message_for_project) { :import_users }
       end
     end
+    context 'when using mistyped mode' do
+      let(:params) do
+        params = {
+          GDC_GD_CLIENT: client,
+          input_source: 'foo',
+          domain: 'bar',
+          gdc_logger: logger,
+          sync_mode: 'unsuported_sync_mode' # sync_one_project_based_on_pid
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      before do
+        allow(domain).to receive(:clients).and_return(organization)
+      end
+      it_should_behave_like 'when using unsuported sync_mode'
+    end
+    context 'when using no mode' do
+      let(:params) do
+        params = {
+          GDC_GD_CLIENT: client,
+          input_source: 'foo',
+          domain: 'bar',
+          gdc_logger: logger
+          # sync_mode: 'unsuported_sync_mode' # sync_one_project_based_on_pid
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      before do
+        allow(domain).to receive(:clients).and_return(organization)
+      end
+      before do
```

Comment:
Identical code found in shared_examples_for_user_actions.rb:29, synchronize_users_spec.rb:165 (mass = 180)